packages/shared/src/components/MainLayout.tsx

```diff
@@ -57,42 +55,44 @@ export default function MainLayout({
   const { windowLoaded } = useContext(ProgressiveEnhancementContext);
   const { user, showLogin, loadingUser } = useContext(AuthContext);
   const [showGreeting, setShowGreeting] = useState(false);
-  const showSidebar = useMedia(
-    [footerNavBarBreakpoint.replace('@media ', '')],
-    [true],
-    false,
-  );
+  const { showSidebar } = useShowSidebar();
   const [openMobileSidebar, setOpenMobileSidebar] = useState(false);
 
   return (
     <>
       <PromotionalBanner />
       <header className="flex relative laptop:fixed laptop:top-0 laptop:left-0 z-3 flex-row-reverse laptop:flex-row justify-between items-center py-3 px-4 tablet:px-8 laptop:px-4 laptop:w-full h-14 border-b bg-theme-bg-primary border-theme-divider-tertiary">
         {!showSidebar && <MobileHeaderRankProgress />}
-        <LinkWithTooltip
-          href={process.env.NEXT_PUBLIC_WEBAPP_URL}
-          passHref
-          prefetch={false}
-          tooltip={{ placement: 'right', content: 'Home' }}
-        >
-          {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions */}
-          <a className="flex items-center" onClick={onLogoClick}>
-            <Logo className={styles.homeSvg} />
-            <CSSTransition
-              in={!showGreeting}
-              timeout={500}
-              classNames="fade"
-              unmountOnExit
+        {mobileTitle && (
+          <h3 className="block laptop:hidden typo-callout">{mobileTitle}</h3>
+        )}
+        {!mobileTitle ||
+          (showSidebar && (
+            <LinkWithTooltip
+              href={process.env.NEXT_PUBLIC_WEBAPP_URL}
+              passHref
+              prefetch={false}
+              tooltip={{ placement: 'right', content: 'Home' }}
             >
-              <LogoText
-                className={classNames(
-                  styles.homeSvg,
-                  'hidden ml-1 laptop:block',
-                )}
-              />
-            </CSSTransition>
-          </a>
-        </LinkWithTooltip>
+              {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions */}
+              <a className="flex items-center" onClick={onLogoClick}>
+                <Logo className={styles.homeSvg} />
+                <CSSTransition
+                  in={!showGreeting}
+                  timeout={500}
+                  classNames="fade"
+                  unmountOnExit
+                >
+                  <LogoText
+                    className={classNames(
+                      styles.homeSvg,
+                      'hidden ml-1 laptop:block',
+                    )}
+                  />
+                </CSSTransition>
+              </a>
+            </LinkWithTooltip>
+          ))}
```

Comment:
This can be extracted to a standalone component I guess.
I know it has been there for long time but it's time to make it a bit cleaner